### ISSUE TITLE
wiki: CodeReviewComments change

### ISSUE THREAD

**Issue by eslam-mahmoud** [golang/go#33228]
Hello, 

I suggest an example to be added to explain why interface constructor should return the struct not the interface type itself

The edit to be added on [CodeReviewComments #interfaces](https://github.com/golang/go/wiki/CodeReviewComments/b61e691f208b5b3523cc3fde1870e8ccec4b2170#interfaces)

`Go interfaces generally belong in the package that uses values of the interface type, not the package that implements those values. The implementing package should return concrete (usually pointer or struct) types: that way, new methods can be added to implementations without requiring extensive refactoring, **and this will enable the struct to implement more than one interface and be accepted/evaluated to all of them in different functions** `

The code example 
https://play.golang.org/p/zZU0rFzjuEm
```go
// DONOT DO THIS
package main
import ("io")
type Thinger interface { 
	Thing() bool 
}
type defaultThinger struct{  }
func (t defaultThinger) Thing() bool { 
	return true 
}
func (t defaultThinger) Read(p []byte) (n int, err error) {
   	return 0, nil
}  	
func NewThinger() Thinger { 
	return defaultThinger{  } 
}
func main() {
	testThinger(NewThinger())
	// will return cannot use NewThinger() (type Thinger) as type io.Reader in argument to testReader: Thinger does not implement io.Reader (missing Read method)
	// if NewThinger() return defaultThinger it will work
	testReader(NewThinger())
}
func testReader(r io.Reader) {}
func testThinger(t Thinger) {}
```

**Comment by ianlancetaylor** [comment]
Honestly, that seems like a minor point to me, compared to the existing point.

CC @bcmills

**Comment by eslam-mahmoud** [comment]
The point is to clarify the why with an example of what can go wrong

**Comment by bcmills** [comment]
Implementing a second interface seems like a special case of the general problem of wanting to access methods beyond the first interface.

In particular, it would seem to suggest that you could instead return an interface that composes together the various other interfaces that you intend to implement. But that would still have the same problems as returning an interface in general: it still would not allow you to add methods beyond the original interface, and still would not allow clarifying documentation on the method implementations.

**Comment by eslam-mahmoud** [comment]
I agree with you, and because I faced that case I tried to add it for everyone to be one more point listed explaining why not to return the interface

**Comment by seankhliao** [comment]
Seems we aren't going to make this change.
